docs/tutorial/typesfuns.rst

```diff
@@ -1182,3 +1183,39 @@ matching ``let`` and lambda bindings. It will *only* work if:
 - The type of the result is "known". i.e. the type of the expression
   can be determined *without* type checking the ``case``-expression
   itself.
+
+Totality
+========
+
+A *total* function is a function that terminates for all possible
+inputs, or one that is guaranteed to produce some output before making
+a recursive call. For example, a function that squares an integer is
+total for all integers:
+
+.. code-block:: idris
+
+    square : Int -> Int
+    square x = x * x
+
+However, we might imagine a function, ``unsafeHead``, that is identical to
+Idris' ``head`` function except that it is *not* total: it will error out
+at runtime if called on an empty list. (Note that unlike Haskell's ``head``,
+Idris' ``head`` function *is* total, as it will fail at compile time rather
+than runtime.) ``unsafeHead`` might look like this:
+
+.. code-block:: idris
+    -- Unsafe head example!
+    unsafeHead : List a -> a
+    unsafeHead (x::xs) = x
+
+And although it typechecks and compiles, it will not reduce:
```

Comment:
Does the reader know about reduction here...we may need to add a short sentence of what that is.